skimage/transform/_warps_cy.pyx

```diff
@@ -110,6 +127,14 @@ def _warp_fast(cnp.ndarray image, cnp.ndarray H, output_shape=None,
     cdef Py_ssize_t rows = img.shape[0]
     cdef Py_ssize_t cols = img.shape[1]
 
+    cdef void (*transform_func)(double, double, double*, double*, double*) nogil
+    if (abs(M[2, 0]) < 1e-10 and
```

Comment:
This seems like a very dangerous approach; e.g., in my work we often optimize over the parameters of a transformation---don't want to be hitting a wall at some arbitrary value.  We should check for absolute zero here.